scripts/reindex.py

```diff
@@ -96,8 +106,31 @@ def main(argv: List[str]):
 
     configure_script_logging(logger)
 
+    source_globs = set(args.sources)
+    if source_globs != {'*'} and not args.partition_prefix_length:
+        parser.error('Cannot specify sources when performing a local reindex')
+
     azul = AzulClient(num_workers=args.num_workers)
 
+    sources_by_catalog = defaultdict(set)
+    globs_matched = set()
+    for catalog in args.catalogs:
+        sources = azul.catalog_sources(catalog)
+        for source_glob in source_globs:
+            matches = fnmatch.filter(sources, source_glob)
+            if matches:
+                globs_matched.add(source_glob)
+            logger.debug('Matched sources %r in catalog %r', matches, catalog)
+            sources_by_catalog[catalog].update(matches)
+    unmatched = source_globs - globs_matched
+    if unmatched:
+        logger.warning('Source(s) not found in any catalog: %r', unmatched)
```

Comment:
```suggestion
```

It's not a problem to for a catalog not to match. Note even a potential one.

Reply:
@hannes-ucsc: "I misunderstood, ignore the diff suggestion."